### PR TITLE
[Pal/Linux-SGX] Improve SGX driver header configuration

### LINKED ISSUE #120: [Docs] Quick Start guide is missing the documentation for the latest meson commands involving -Dsgx_driver and -Dsgx_driver_path
## Description of the problem

Building page is updated with the new parameters but the same is missing in the Quick Start guide.

https://gramine.readthedocs.io/en/latest/building.html#id1

https://gramine.readthedocs.io/en/latest/quickstart.html#quick-start-with-sgx-support

### PR DESCRIPTION
**Pull request by pwmarcz**
## Description of the changes 

This change improves support for building on systems other than Ubuntu:

- Use `asm/sgx.h` (not `uapi/asm/sgx.h`). Previously, `uapi` was always appended to the provided include path, so it wasn't possible to configure Gramine when the userspace kernel headers were installed without `uapi` prefix.

- Try out the `/usr/include` path. Ubuntu doesn't install kernel headers there, but many other distributions (e.g. Fedora, CentOS) do.

- Clarify in the quick start guide that it is for Ubuntu, and add a hint in case of configuration problems.

Closes #120.

## How to test this PR? 

Try building on a system with 5.11+ kernel.

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @pwmarcz)

___
*[Documentation/building.rst, line 181 at r1](https://reviewable.io/reviews/gramineproject/gramine/121#-MlADWAl4KtLe-TtzV0j:-MlADWAl4KtLe-TtzV0k:bjd6mwa) ([raw file](https://github.com/gramineproject/gramine/blob/926f807eaf4d0dfc5bac392301f16739ee5f574f/Documentation/building.rst#L181)):*
> ```reStructuredText
> where the SGX driver was downloaded or installed in the previous step. For
> example, for the DCAP version 33 of the SGX driver, you must specify
> ``-Dsgx_driver_include_path="/usr/src/sgx-1.33/"``. If this parameter is
> ```

On my Dcap-sgx-driver system, it should be `/usr/src/sgx-1.41/include`. So please append `include` to the path.

**Review by pwmarcz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: 2 of 3 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL), "fixup! " found in commit messages' one-liners (waiting on @dimakuv)

___
*[Documentation/building.rst, line 181 at r1](https://reviewable.io/reviews/gramineproject/gramine/121#-MlADWAl4KtLe-TtzV0j:-MlAKtY_CN-X-khzVYqE:b-oatshc) ([raw file](https://github.com/gramineproject/gramine/blob/926f807eaf4d0dfc5bac392301f16739ee5f574f/Documentation/building.rst#L181)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

On my Dcap-sgx-driver system, it should be `/usr/src/sgx-1.41/include`. So please append `include` to the path.
</blockquote></details>

Done. And I also changed the example version to 1.41 (it's actually 1.41, not 41, right?)

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), "fixup! " found in commit messages' one-liners

___
*[Documentation/building.rst, line 181 at r1](https://reviewable.io/reviews/gramineproject/gramine/121#-MlADWAl4KtLe-TtzV0j:-MlANeHK77AMB7ZKarY5:b6p54fl) ([raw file](https://github.com/gramineproject/gramine/blob/926f807eaf4d0dfc5bac392301f16739ee5f574f/Documentation/building.rst#L181)):*
<details><summary><i>Previously, pwmarcz (Paweł Marczewski) wrote…</i></summary><blockquote>

Done. And I also changed the example version to 1.41 (it's actually 1.41, not 41, right?)
</blockquote></details>

Yes, this change is fine.

The versioning in the DCAP driver is a bit weird: https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/driver/linux#change-log. The SGX DCAP driver is bundled together with the rest of the DCAP SGX libraries. So you see GitHub repo tags like DCAP v1.12. But the corresponding version of the SGX DCAP driver (which tracked the in-kernel SGX driver development) is v1.41. And this SGX DCAP driver v1.41 tracks the revision 41 of the in-kernel SGX driver (in the Linux kernel mailing list).

Basically, when people talk about the SGX DCAP driver v1.12 or v1.41 or v41, they mean the same thing :)

**Review by mkow — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: all files reviewed, all discussions resolved, "fixup! " found in commit messages' one-liners

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/121)* status: :shipit: complete! all files reviewed, all discussions resolved
